### PR TITLE
Virtual node taints

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr adds the possibility to specify additional annotations and taint to be applied in the liqo node setting them in the VirtualNode spec

# How Has This Been Tested?

- [x] locally

### CODE REVIEW

File: pkg/utils/slice/slice.go [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
 package slice
 
-// ContainsString checks if a given slice of strings contains the provided string.
-func ContainsString(slice []string, s string) bool {
+// Contains checks if a given slice contains the provided element.
+func Contains[K comparable](slice []K, s K) bool {
```
Comment (fra98) [inline pkg/utils/slice/slice.go:18]: I think we could delete this function and use the new Contains() from standard library package slices

File: pkg/utils/slice/slice.go [hunk 2]
```diff
@@ -24,10 +24,10 @@ func ContainsString(slice []string, s string) bool {
 	return false
 }
 
-// RemoveString returns a newly created []string that contains all items from slice that
+// Remove returns a newly created slice that contains all items from slice that
 // are not equal to s.
-func RemoveString(slice []string, s string) []string {
-	newSlice := make([]string, 0)
+func Remove[K comparable](slice []K, s K) []K {
+	newSlice := make([]K, 0)
```
Comment (fra98) [inline pkg/utils/slice/slice.go:30]: What about using standard library function `DeleteFunc()` ?
```suggestion
func Remove[K comparable](slice []K, s K) []K {
	return slices.DeleteFunc(slice, func(t K) bool {
		return t == s
	})
}
```

File: pkg/virtualKubelet/liqoNodeProvider/setup.go [hunk 3]
```diff
@@ -101,7 +101,7 @@ func node(cfg *InitConfig) *corev1.Node {
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        cfg.NodeName,
 			Labels:      labels.Merge(lbls, cfg.ExtraLabels),
-			Annotations: cfg.ExtraAnnotations,
+			Annotations: cfg.ExtraAnnotations, // TODO: check that they are not overwritten by annotations patch function
```
Comment (fra98) [inline pkg/virtualKubelet/liqoNodeProvider/setup.go:104]: Is this TODO still relevant?

**Review by fra98 — APPROVED** [review]

File: pkg/virtualKubelet/liqoNodeProvider/setup.go (hunk 3 above)
Comment (aleoli) [reply pkg/virtualKubelet/liqoNodeProvider/setup.go:104]: No, I checked it, but forgot to remove the comment

**Pull request merged**